tool/cli.js

```diff
@@ -504,25 +499,35 @@ program
         fixFlaws: true,
         fixFlawsDryRun: true,
       });
-
-      const flaws = Object.values(doc.flaws || {})
-        .map((a) => a.filter((f) => f.fixable).length || 0)
-        .reduce((a, b) => a + b, 0);
-      if (flaws === 0) {
-        console.log(chalk.green("Found no fixable flaws!"));
+      let flaws = Object.values(doc.flaws || {}).flat();
+      const flawCount = flaws.length;
+      if (flawCount === 0) {
+        console.log(chalk.green("✓ All seems fine"));
         return;
       }
-      const { run } = yes
-        ? { run: true }
-        : await prompt({
-            type: "confirm",
-            message: `Proceed fixing ${flaws} flaws?`,
-            name: "run",
-            default: true,
+      const nonFixableFlaws = flaws.filter((f) => !f.fixable);
```

Comment:
There are some types of flaw that does not contain `fixable` field. They are `bad_bcd_queries`, `bad_bcd_links`, and `sectioning`. It seems that these three types of flaw can not be fixed automatically, so this line of code should work. However, maybe we need to unify different types of flaw object. Currently, there's no any JSON schema or type declaration for flaws.

Reply:
> @peterbe: both of which don't have a natural way to automate fixing.

For `bad_bcd_queries` and `bad_bcd_links` flaw, @peterbe mentioned that in https://github.com/mdn/yari/issues/680#issuecomment-660151884

For `sectioning` flaw, there are three types of it right now:

1. [countBCDDataDivsFound !== countPotentialBCDDataDivs](https://github.com/LuckyWindsck/yari/blob/correct-fix-flaws-logic/build/document-extractor.js#L204-L210)
```javascript
if (countBCDDataDivsFound !== countPotentialBCDDataDivs) {
  const leftoverCount = countPotentialBCDDataDivs - countBCDDataDivsFound;
  const explanation = `${leftoverCount} 'div.bc-data' element${
    leftoverCount > 1 ? "s" : ""
  } found but deeply nested.`;
  flaws.push(explanation);
}
```

2. [Excess \<h2\> tag that is NOT at root-level](https://github.com/LuckyWindsck/yari/blob/correct-fix-flaws-logic/build/document-extractor.js#L368-L376)
```javascript
for (const i of [...Array(h2s.length).keys()]) {
  if (i) {
    // Excess!
    flaws.push(
      `Excess <h2> tag that is NOT at root-level (id='${h2s
        .eq(i)
        .attr("id")}', text='${h2s.eq(i).text()}')`
    );
  } ...
```
3. [Excess \<h3\> tag that is NOT at root-level](https://github.com/LuckyWindsck/yari/blob/correct-fix-flaws-logic/build/document-extractor.js#L389-L397)
```javascript
for (const i of [...Array(h3s.length).keys()]) {
  if (i) {
    // Excess!
    flaws.push(
      `Excess <h3> tag that is NOT at root-level (id='${h3s
        .eq(i)
        .attr("id")}', text='${h3s.eq(i).text()}')`
    );
  } else {
```

Since I am new to this yari project, I am not familiar with the flaw system. However I think that none of them can be automatically fixed in my opinion.

Reply:
I studied the code and made the following summary of all types of flaws.

1. Whether this type of flaw contains `fixable` field?
2. Whether it will be fixed in the [fixFixableFlaws](https://github.com/LuckyWindsck/yari/blob/correct-fix-flaws-logic/build/flaws/index.js#L73-L341) function?
3. Is it possible for the value of `fixable` be true?

| Flaw type                 | Q1  | Q2  | Q3  | flaw declaration            | fixable declaration              |
| ------------------------- | --- | --- | --- | --------------------------- | -------------------------------- |
| doc.flaws.macros          | √   | √   | √   | build/index.js#L321         | build/index.js#L339              |
| doc.flaws.broken_links    | √   | √   | √   | build/flaws/index.js#L43    | build/flaws/broken-links.js#L98  |
| doc.flaws.bad_bcd_queries | x   | x   | ?   | build/flaws/index.js#L43    |                                  |
| doc.flaws.bad_bcd_links   | x   | x   | ?   | build/bcd-urls.js#L18       |                                  |
| doc.flaws.images          | √   | √   | √   | build/check-image.js#L47    | build/check-image.js#L59         |
| doc.flaws.image_widths    | √   | √   | √   | build/check-image.js#L256   | build/check-image.js#L283        |
| doc.flaws.bad_pre_tags    | √   | √   | √   | build/flaws/index.js#L43    | build/flaws/pre-tags.js#L53      |
| doc.flaws.sectioning      | x   | x   | ?   | build/flaws/sections.js#L12 |                                  |
| doc.flaws.heading_links   | √   | x   | x   | build/flaws/index.js#L43    | build/flaws/heading-links.js#L47 |
| doc.flaws.unsafe_html     | √   | x   | x   | build/flaws/index.js#L43    | build/flaws/unsafe-html.js#L43   |

Reply:
I think it's safe to assume that if `flaw.fixable === undefined` it's the same as `flaw.fixable === false`. 
We don't have a schema or TypeScript on this area of the code :( It has grown too organically. 
What I think would be an easy and cheap start towards that is to put in `.fixable = false` in those that don't already have the key at all. Would you mind doing that?

Reply:
> Since I am new to this yari project, I am not familiar with the flaw system. However I think that none of them can be automatically fixed in my opinion.

But I think you've understood it very well. We introduced the `.fixable` a bit late. At some point it used to be that `.suggestion` being truthy was good enough to assume fixable. 

The general idea with a flaw is that they all have an `id` (string) and `explanation` (string). 
You can see it much better here: https://github.com/mdn/yari/blob/2c644475651c5e8f86ed291c7e764dbc0b5acc6a/client/src/document/types.tsx#L8-L15 which is a bit more organized than how it gets created on the Node side.